assets/javascripts/discourse/components/global-filter-item.js

```diff
@@ -13,6 +14,6 @@ export default Component.extend({
 
   @action
   selectFilter(tag) {
-    DiscourseURL.routeTo(`/tag/${tag}`);
+    this.router.transitionTo(`/categories?tag=${tag}`);
```

Comment:
Note, we're assuming that `/categories` is the default homepage. We probably want to also support latest, so we need a check of the top menu site setting here.